kubelet/datadog_checks/kubelet/cadvisor.py

```diff
@@ -0,0 +1,169 @@
+        return requests.get(self.cadvisor_legacy_url, timeout=timeout).json()
+
+    def process_cadvisor(self, instance):
+        self.max_depth = instance.get('max_depth', DEFAULT_MAX_DEPTH)
+        enabled_gauges = instance.get('enabled_gauges', DEFAULT_ENABLED_GAUGES)
+        self.enabled_gauges = ["{0}.{1}".format(NAMESPACE, x) for x in enabled_gauges]
+        enabled_rates = instance.get('enabled_rates', DEFAULT_ENABLED_RATES)
+        self.enabled_rates = ["{0}.{1}".format(NAMESPACE, x) for x in enabled_rates]
+        pod_level_metrics = instance.get('pod_level_metrics', DEFAULT_POD_LEVEL_METRICS)
+        self.pod_level_metrics = ["{0}.{1}".format(NAMESPACE, x) for x in pod_level_metrics]
+
+        self._update_metrics(instance)
+
+    def _update_metrics(self, instance):
+        def parse_quantity(s):
+            number = ''
+            unit = ''
+            for c in s:
+                if c.isdigit() or c == '.':
+                    number += c
+                else:
+                    unit += c
+            return float(number) * FACTORS.get(unit, 1)
+
+        metrics = self.retrieve_cadvisor_metrics()
+
+        if not metrics:
+            raise Exception('No metrics retrieved cmd=%s' % self.metrics_cmd)
+
+        for subcontainer in metrics:
+            c_id = subcontainer.get('id')
+            if 'aliases' not in subcontainer:
+                # it means the subcontainer is about a higher-level entity than a container
+                continue
+            try:
+                self._update_container_metrics(instance, subcontainer)
+            except Exception as e:
+                self.log.error("Unable to collect metrics for container: {0} ({1})".format(c_id, e))
+
+    def _publish_raw_metrics(self, metric, dat, tags, is_pod, depth=0):
+        if depth >= self.max_depth:
+            self.log.warning('Reached max depth on metric=%s' % metric)
+            return
+
+        if isinstance(dat, numbers.Number):
+            # Pod level metric filtering
+            is_pod_metric = False
+            if self.pod_level_metrics and any([fnmatch(metric, pat) for pat in self.pod_level_metrics]):
+                is_pod_metric = True
+            if is_pod_metric != is_pod:
+                return
+
+            # Metric submission
+            if self.enabled_rates and any([fnmatch(metric, pat) for pat in self.enabled_rates]):
+                self.rate(metric, float(dat), tags)
+            elif self.enabled_gauges and any([fnmatch(metric, pat) for pat in self.enabled_gauges]):
+                self.gauge(metric, float(dat), tags)
+
+        elif isinstance(dat, dict):
+            for k, v in dat.iteritems():
+                self._publish_raw_metrics(metric + '.%s' % k.lower(), v, tags, is_pod, depth + 1)
+
+        elif isinstance(dat, list):
+            self._publish_raw_metrics(metric, dat[-1], tags, is_pod, depth + 1)
+
+    def _update_container_metrics(self, instance, subcontainer):
+        tags = []
```

Comment:
`tags` definition isn't needed as you are redefining it in each condition below.